### PR TITLE
Fix Codelite/gmake2 `buildmessage` escaping.

### PR DESCRIPTION
**Pull request by Jarod42**
**What does this PR do?**

Fix escaping for `buildmessage` for gmake2 and codelite

**How does this PR change Premake's behavior?**

Escape double quote and backslash in `quoted`

**Anything else we should know?**

Tested with https://github.com/Jarod42/premake-sample-projects/actions with branch custom_rule_check_quote_in_message

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [ ] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [ ] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

**Review by nickclark2016 — APPROVED** [review]
LGTM

File: modules/codelite/tests/test_codelite_additional_rules.lua [hunk 1]
```diff
@@ -60,7 +60,7 @@
 			value = 0
 		}
 
-		buildmessage 'Rule-ing %{file.name}'
+		buildmessage "Rule'ing %{file.name}"
```
Comment (samsinsane) [inline modules/codelite/tests/test_codelite_additional_rules.lua:63]: Why did `-` change to `'`? Seems like an unintended/unnecessary change?

File: modules/codelite/tests/test_codelite_additional_rules.lua [hunk 2]
```diff
@@ -213,15 +213,15 @@ test2.obj: test2.rule
 		buildinputs { "toto.txt", "extra_dependency" }
 		buildoutputs { "toto.c" }
 		buildcommands { "test", "test toto.c" }
-		buildmessage "Some message"
+		buildmessage '"Some message"'
```
Comment (samsinsane) [inline modules/codelite/tests/test_codelite_additional_rules.lua:216]: You should avoid hijacking existing tests, you should create a new test explicitly for testing the escaping.

**Review by samsinsane — CHANGES_REQUESTED** [review]
gmake2 also needs a new test for the escaping.

File: modules/codelite/tests/test_codelite_additional_rules.lua (hunk 1 above)
Comment (Jarod42) [reply modules/codelite/tests/test_codelite_additional_rules.lua:63]: It was to show/test single quote escaping.
I can revert that part if you prefer.

File: modules/codelite/tests/test_codelite_additional_rules.lua (hunk 2 above)
Comment (Jarod42) [reply modules/codelite/tests/test_codelite_additional_rules.lua:216]: OK, I will do that.

File: modules/codelite/tests/test_codelite_additional_rules.lua (hunk 1 above)
Comment (samsinsane) [reply modules/codelite/tests/test_codelite_additional_rules.lua:63]: That should be done in a dedicated test for escaping, as mentioned in my other comment, as this looks like a typo and will likely end up being changed back eventually.

Comment (Jarod42) [reply modules/codelite/tests/test_codelite_additional_rules.lua]: Roll backed.

File: modules/codelite/tests/test_codelite_additional_rules.lua (hunk 2 above)
Comment (Jarod42) [reply modules/codelite/tests/test_codelite_additional_rules.lua:236]: Roll backed.
Dedicated test done.

**Comment by Jarod42** [comment]
> [  FAILED  ] premake_http.https_get_verify_peer
> HTTP response code said error
> The requested URL returned error: 504 Gateway Time-out

Unrelated to my changes. :-)
Not the first time that those tests failed randomly BTW.

**Review by samsinsane — APPROVED** [review]
